.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+    branches:
+      # Branches from forks have the form 'user:branch-name' so we only run
+      # this job on pull_request events for branches that look like fork
+      # branches. Without this we would end up running this job twice for non
+      # forked PRs, once for the push and then once for opening the PR.
+    - '**:**'
+
+jobs:
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+#      - name: Cache Gradle
+#        uses: actions/cache@v2
+#        with:
+#          path: |
+#            ~/.gradle/wrapper
+#            ~/.gradle/caches
+#          key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - id: get_version
+        name: "Get Version"
+        uses: battila7/get-version-action@v2
+      - name: Build with Gradle
+        if: startsWith(github.ref, 'refs/tags/v') != true
+        run: |
+          ./gradlew check
+      - name: Build with Gradle & Release on Tag
+        if: startsWith(github.ref, 'refs/tags/v')
+        env:
+          TAG_NAME: "${{ steps.get_version.outputs.version }}"
+          ORG_GRADLE_PROJECT_bintrayApiKey: ${{ secrets.ORG_GRADLE_PROJECT_BINTRAYAPIKEY }}
+          ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.ORG_GRADLE_PROJECT_SONATYPEPASSWORD }}
+        run: |
+          ./gradlew check
+          ./gradlew release "-PreleaseVersion=${{env.TAG_NAME}}"
```

Comment:
i assume we cannot test this besides actually build a release correct?

Reply:
Yeah, we'll only know everything is correct if it can actually upload to bintray/mavenCentral, so a real release is necessary.

Reply:
@oehme here is a check too in case you did not see